### PR TITLE
Topics/gatsby source medium | add limit

### LINKED ISSUE #6181: [gatsby-source-medium] Allow custom post limit to be set
## Summary

The plugin currently gets the default number of latest posts from Medium but you are able to increase this by passing a `limit` parameter. 

### Basic example

I've not tested this as I'm not certain how you'd implement, but I think it should be a matter of setting the `limit` parameter is this part of `gatsby-node.js`:
```
const fetch = (username, limit = 10) => {
  const url = `https://medium.com/${username}/latest?format=json&limit=${limit}`
  return axios.get(url)
}
```
To supprt the above call you'd need to set the limit parameter in `gatsby-config.js`:
```
// In your gatsby-config.js
plugins: [
  {
    resolve: `gatsby-source-medium`,
    options: {
      username: `username/publication`,
      limit: 100,
    },
  },
]
```
and using it here:
```
exports.sourceNodes = async ({ actions, createNodeId }, { username, limit }) => {
  const { createNode } = actions

  try {
    const result = await fetch(username, limit)
```

### Motivation

I want to be able to load all of my Medium articles to display in my Gatsby site. 

I'd prefer to load them all by default, and then limit the results using the graphql query, but I can't see a way to do that using this endpoint.

### PR DESCRIPTION
**Pull request by seaburrows**
fixes #6181 

**Caveat**
This isn't really tested 😱 

There are no tests within this package and the `examples/using-medium` site is broken on master. After some digging around I can see that the netlify builds are failing with the same error I get: https://app.netlify.com/sites/using-medium/deploys

Since the upgrade has been merged in (https://github.com/gatsbyjs/gatsby/pull/5704) and the existing build is failing **and** I don't know how to fix them, I figured I'd open this PR rather than lose it.

Please let me know if I should be raising any other tickets.

Thanks!

### CODE REVIEW

**Comment by pieh** [comment]
Hey, I will open PR that fixes `gatsby-source-medium@next` (`using-medium` example is for some reason using v1 version of that plugin which works, but it break as soon as we use `gatsby-dev` for it).

**Comment by pieh** [comment]
@seaburrows PR with fix I mentioned in previous comment was merged, so you can update your branch, which should fix `using-medium` preview for this PR

File: examples/using-medium/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "gatsby-example-using-medmium",
+  "name": "gatsby-example-using-medium",
```
Comment (pieh) [inline examples/using-medium/package.json]: 🙈 😄

File: examples/using-medium/gatsby-config.js [hunk 2]
```diff
@@ -6,7 +6,12 @@ module.exports = {
     {
       resolve: `gatsby-source-medium`,
       options: {
+        // To test the limit parameter
+        // Replace the publication username with a user
+        // Limit is optional - set quite low as example user doesn't have that many posts
         username: `smartive`,
+        // username: `@ReactEurope`,
+        // limit: 20, // note: the graphql query in src/pages/index.js also imposes a limit
```
Comment (pieh) [inline examples/using-medium/gatsby-config.js]: commeting this one out is intentional?

Comment (pieh) [reply examples/using-medium/gatsby-config.js]: I would think that just documenting it in plugin docs would be enough, but we can keep it here

**Comment by pieh** [comment]
generally this is fine - just couple DX related suggestions:
 - we could validate that `limit` is a postive integer number
 - (not sure) we could display warning that `limit` is only useful for user - this is me guessing - if username doesn't start with `@` (this would mean that it's user, right?) and `limit` is defined show warning that `limit` has no effect

**Comment by seaburrows** [comment]
ah okay sure, we can do those! 
I guess then we'd also need to check that `username` is a string and is not empty?

So it would be:

- show an error if `username` is not a string or is an empty string
- show an error if limit is not a positive integer number (or just ignore it?)
- show a warning if the limit has been set, but the `username` starts with `@`

**Comment by pieh** [comment]
> show an error if username is not a string or is an empty string
> show an error if limit is not a positive integer number (or just ignore it?)
> show a warning if the limit has been set, but the username starts with @

Sounds great!

**Comment by calcsam** [comment]
@seaburrows -- would you be able to make changes discussed? we'd love to get this in!

**Comment by m-allanson** [comment]
@pieh how do you feel about merging this in, then seeing if anyone wants to follow up with your suggested changes?

**Comment by pieh** [comment]
Yeah, let's merge this in.

**Comment by seaburrows** [comment]
whoops, sorry for long silence. I've been super swamped. Glad this ended up getting in!! 🎉 😁 
Hopefully life calms down a bit soon and I can pick up any remaining work 😅

**Pull request merged**